src/lib/support/logging/CHIPLogging.cpp

```diff
@@ -102,10 +102,30 @@ static const char ModuleNames[] = "-\0\0" // None
 
 void GetModuleName(char * buf, uint8_t bufSize, uint8_t module)
 {
-    const char * moduleNamePtr = ModuleNames + ((module < ModuleNamesCount) ? module * chip::Logging::kMaxModuleNameLen : 0);
+    if (bufSize == 0)
+    {
+        return; // no space for anything
+    }
```

Comment:
We can't just return here.  The caller is likely expecting a null-terminated string.  If we can't do that, we need to propagate an error to the caller.

Why don't we define and pass a CapacityBoundBuffer<length> type or similar so this code can know at compile time the size of the buffer it's going to write into?

Reply:
Should I die insead? I am being told that there is no space in the buffer, not even for null terminator.

Reply:
Also, doing what snprintf does for this case seems like a reasonable approach given the external interface of this function.

Reply:
Updated to enforce at compile time.